packages/analytics-js-integrations/src/integrations/Mixpanel/util.js

```diff
@@ -32,7 +33,71 @@ const traitAliases = {
   phone: '$phone',
 };
 
-const formatTraits = message => {
+/**
+ * Removes a property from an object based on a given property path.
+ * 
+ * @param {object} obj - The object from which the property needs to be removed.
+ * @param {string} propertyPath - The path of the property to be removed, using dot notation.
+ * @returns {undefined} - This function does not return anything.
+ *
+ * @example
+ * const obj = {
+ *   person: {
+ *     name: 'John',
+ *     age: 30,
+ *     address: {
+ *       city: 'New York',
+ *       state: 'NY'
+ *     }
+ *   }
+ * };
+ * 
+ * unset(obj, 'person.address.city');
+ *  Output: { person: { name: 'John', age: 30, address: { state: 'NY' } } }
+ */
+function unset(obj, propertyPath) {
+  const keys = propertyPath.split('.');
+  const lastKey = keys.pop();
+
+  let current = obj;
+  for (const key of keys) {
+    if (current[key] === undefined || current[key] === null) {
+      return; // Property path not valid, nothing to unset
+    }
+    current = current[key];
+  }
+
+  delete current[lastKey];
+}
+
+function filterSetOnceTraits(outgoingTraits, setOnceProperties) {
+  // Create a copy of the original traits object
+  const traitsCopy = { ...outgoingTraits };
+
+  // Initialize setOnce object
+  const setOnceEligible = {};
+
+  // Step 1: find the k-v pairs of setOnceProperties in traits and contextTraits
+
+  setOnceProperties.forEach((propertyPath) => {
+    const pathSegments = propertyPath.split('.');
+    const propName = pathSegments[pathSegments.length - 1];
+
+    if (Object.keys(traitsCopy).length > 0 && get(traitsCopy, propertyPath)) {
+      setOnceEligible[propName] = get(traitsCopy, propertyPath);
```

Comment:
As the propertyPath will contain the exact path of the trait can we run it on top of the payload apart from running on traitsCopy?

Reply:
no we need not. Reason being :
-  this is identify event so we will expect traits only not properties. 
-  If we follow our already [existing code](https://github.com/rudderlabs/rudder-sdk-js/blob/89b216283752b53d272e3015dc7b644f94799211/packages/analytics-js-integrations/src/integrations/Mixpanel/util.js#L101) we are just fetching traits from context only. 
- The property path concept has come for any sub-properties, like for `address.city` and `address.country` etc where we need the exact path of the particular property in payload to fetch its values. Will raise doc card accordingly